### PR TITLE
Add support for accessing values from data-structures contained within `Const`

### PR DESCRIPTION
**Pull request by kmr-srbh**
# Overview
Trying to access a key from a `Const dict` threw a `SemanticError` stating type mismatch. An `ASR pass error` was thrown when trying to access values from a `Const list`, `Const tuple` and `Const str`. This was chiefly because the case of a `Const` value was not addressed in handling subscript indices.

- Screenshot from 2024-02-26 12-21-43

- Screenshot from 2024-02-29 19-25-04

# Fix
- Check for `Const` when visiting subscript indices and handle item access through the _contained type_ in the `llvm` back-end.
- Throw a `SemanticError` for immutable types like `tuple` and `str`.

- ## Tuple
Screenshot from 2024-02-29 20-27-55
- ## String
Screenshot from 2024-02-29 20-37-10
- ## Dictionary
Screenshot from 2024-02-29 20-28-35
- ## List
Screenshot from 2024-02-29 20-29-00

# Note
The `get` attribute of `Const dict` still throws a `SemanticError` now. I will push the fix for that in a separate PR.

### CODE REVIEW

**Comment by kmr-srbh** [comment]
It is okay for the tests to fail as the PR does contain a bug related to `value` as stated above.

**Comment by kmr-srbh** [comment]
The failing tests are irrelevant to the changes made.

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -7194,57 +7267,6 @@ we will have to use something else.
                 tmp = ASR::make_LogicalConstant_t(al, loc, is_space,
                         ASRUtils::TYPE(ASR::make_Logical_t(al, loc, 4)));
                 return;
-            } else if (attr_name == "isalpha") {
```
Comment (Thirumalai-Shaktivel) [inline src/lpython/semantics/python_ast_to_asr.cpp:7197]: The CI fails because of the following changes, 
Why it needs to be modified?

**Comment by Thirumalai-Shaktivel** [comment]
Please mark this PR ready for review once it is ready.

File: src/lpython/semantics/python_ast_to_asr.cpp (hunk 1 above)
Comment (kmr-srbh) [reply src/lpython/semantics/python_ast_to_asr.cpp:7197]: My bad @Thirumalai-Shaktivel. I made an error while creating this branch. These are somebody else's changes. I am opening a new PR as a change needs to be made regarding the method I use for checking `Const` types.

**Pull request closed without merging**